pandas/tests/groupby/test_groupby.py

```diff
@@ -5777,6 +5769,19 @@ def test_agg_over_numpy_arrays(self):
 
         assert_frame_equal(result, expected)
 
+    def test_cummin_cummax(self):
+        df = pd.DataFrame({'A': [1, 1, 1, 2, 2, 2],
+                           'B': range(3, 9)})
+        result = df.groupby('A').cummin()
+        expected = pd.DataFrame({'B': [3, 3, 3, 6, 6, 6]})
+        tm.assert_frame_equal(result, expected)
+
+        df = pd.DataFrame({'A': [1, 1, 1, 2, 2, 2],
+                           'B': range(8, 2, -1)})
```

Comment:
needs tests with nans and different dtypes

needs test against doing .apply(lambda x: x.cummin()) for validation